### ISSUE TITLE
Add `awaitFirstOrNull` to Rx integration modules

### ISSUE THREAD

**Issue by elizarov** [Kotlin/kotlinx.coroutines#224]
See this SO question: https://stackoverflow.com/questions/48617490/how-to-handle-the-null-scenario-with-kotlinx-coroutines-reactive

The Kotlin-style solution would be to have `awaitFirstOrNull` (similar to the naming of the corresponding functions in Kotlin stdlib), so that one can write:

```
val person = peopleRepository.findById(personId).awaitFirstOrNull() ?: doSomethingIfNotFund()
```

**Comment by konrad-kaminski** [comment]
I'd also add `awaitFirstOrElse(defaultValue: () -> T): T`.

**Comment by elizarov** [comment]
Indeed, in Kotlin stdlib there is `getOrDefault` for maps, but that stems from the fact that maps can have null values. Since rective streams don't support nulls I don't see much advange in having`awaitFirstOrElse { something }` when we can do `awaitFirstOrNull() ?: something` for the same effect. Am I missing something?

**Comment by konrad-kaminski** [comment]
There are two reasons I can think of that make the approach with all three of `awaitFirstOrNull/awaitFirstOrDefault/awaitFirstOrElse` functions necessary:
1. Rx1 Observable supports `null` values (I actually added a test case for that in the PR - search for `testObservableWithNull`). This was [removed in Rx2](https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#nulls). So we may either have those functions only for Rx1 or for consistency have them for all reactive types.
2.  `awaitFirstOrNull() ?: something` has higher cognitive load than `awaitFirstOrDefault(something)` or `awaitFirstOrElse { something() }`. If you use `awaitFirstOrNull() ?: something` you actually use `null` to indicate that the stream is empty and then based on that you provide a default. With the other functions you explicitly state the default - there is no indirection, no intermediate "marker" value. IMHO it makes the code also more readable as the intent is explicit.

**Comment by elizarov** [comment]
Released in version 0.22.2
